src-tauri/src/commands.rs

```diff
@@ -714,27 +671,30 @@ pub async fn get_transaction_history(
 pub async fn import_seed_words(
     seed_words: Vec<String>,
     _window: tauri::Window,
-    state: tauri::State<'_, UniverseAppState>,
     app: tauri::AppHandle,
 ) -> Result<(), String> {
     let timer = Instant::now();
     let config_path = app
-        .path_resolver()
+        .path()
         .app_config_dir()
         .expect("Could not get config dir");
     let data_dir = app
-        .path_resolver()
+        .path()
         .app_local_data_dir()
         .expect("Could not get data dir");
 
-    stop_all_processes(state.inner().clone(), false).await?;
+    stop_all_processes(app.clone(), false).await?;
 
     tauri::async_runtime::spawn(async move {
         match InternalWallet::create_from_seed(config_path, seed_words).await {
             Ok(_wallet) => {
-                InternalWallet::clear_wallet_local_data(data_dir).await?;
-                info!(target: LOG_TARGET, "[import_seed_words] Restarting the app");
-                app.restart();
+                if InternalWallet::clear_wallet_local_data(data_dir)
```

Comment:
`InternalWallet::clear_wallet_local_data(data_dir).await?;` I would stick to this or handle `false case` when `InternalWallet::clear_wallet_local_data` return error as now with the if case we do nothing in this situation|  `?` at the end returns the error, its shorter version for writing
`Ok(val) => val`
`Err(e) => return Err(e)`

Reply:
@Misieq01 This whole function called by the invoke is a bit strange now, as I don't think the spawned thread will return and of the results. Which isn't a problem, but is causing issue for the compiler. Can you take a closer look at fix it up a bit?

Reply:
Hmm,so maybe it would be best to return to initial shanon solution

Reply:
I can look at it after my lunch break